feat: Add TopBar component and icons

Mostafa-alsaygh opened the pull request:
add top bar design component 
![image](https://github.com/user-attachments/assets/329c271f-650e-416c-90f2-5011d5c15496)

Mostafa-alsaygh committed:
feat: Add TopBar component and icons

This commit introduces a new `TopBar` composable function and its associated icons: `add_icon.xml`, `arrow_left.xml`, `icondesign.xml`, and `pencil_edit.xml`.

The `TopBar` component provides different variations based on the `TopBarType` enum, allowing for default, app bar, one-option, two-option, without back arrow, and without screen title configurations.

The new icons are used within the `TopBar` component for navigation and actions.

Mostafa-alsaygh committed:
Merge branch 'develop' into component/top-bar

# Conflicts:
#	designSystem/src/main/java/com/london/designsystem/component/TopBar.kt

yusufnasserdev reviewed (changes requested):
Move these files to the system design module.

Abdulrahman-Ragab-01 commented on app/src/main/java/com/london/novix/presentation/component/TopBar.kt:
```diff
@@ -0,0 +1,149 @@
+    Row(
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(horizontal = 16.dp, vertical = 12.dp),
```
vertical padding is equal 8.dp in figma

Abdulrahman-Ragab-01 commented on app/src/main/java/com/london/novix/presentation/component/TopBar.kt:
```diff
@@ -0,0 +1,149 @@
+
+
+@Composable
+@Preview(
```
use our custom preview @ThemePreviews

Abdulrahman-Ragab-01 commented on app/src/main/java/com/london/novix/presentation/component/TopBar.kt:
```diff
@@ -0,0 +1,149 @@
+                    )
+                    Column {
+                        Text(
+                            text = "Novix",
```
move the strings to string file resource

Abdulrahman-Ragab-01 commented on app/src/main/java/com/london/novix/presentation/component/TopBar.kt:
```diff
@@ -0,0 +1,149 @@
+                            color = NovixTheme.colors.body
+                        )
+                        Text(
+                            text = "Born from Nova, Made for Flix.",
```
move the strings to string file resource

Abdulrahman-Ragab-01 commented on app/src/main/java/com/london/novix/presentation/component/TopBar.kt:
```diff
@@ -0,0 +1,149 @@
+            Text(
+                text = title,
+                style = NovixTheme.typography.title.large,
+                color = NovixTheme.colors.body,
```
the text color = NovixTheme.colors.title not body

Abdulrahman1890 commented on app/src/main/java/com/london/novix/presentation/component/TopBar.kt:
```diff
@@ -0,0 +1,149 @@
+                text = title,
+                style = NovixTheme.typography.title.large,
+                color = NovixTheme.colors.body,
+                modifier = Modifier.weight(1f).padding(horizontal = 4.dp)
```
the horizontal padding is equal 12 not 4

Mostafa-alsaygh committed:
feat: Add TopBar component and icons

This commit introduces a new `TopBar` composable function and its associated icons: `add_icon.xml`, `arrow_left.xml`, `icondesign.xml`, and `pencil_edit.xml`.

The `TopBar` component provides different variations based on the `TopBarType` enum, allowing for default, app bar, one-option, two-option, without back arrow, and without screen title configurations.

The new icons are used within the `TopBar` component for navigation and actions.

Additionally, new string resources `appName` and `explainTheNameOfApp` have been added in both English and Arabic.

Mostafa-alsaygh committed:
Merge branch 'develop' into component/top-bar

# Conflicts:
#	designSystem/src/main/res/values-ar/strings.xml
#	designSystem/src/main/res/values/strings.xml

Mostafa-alsaygh committed:
refactor: Change package name for TopBar component

Mostafa-alsaygh committed:
feat: Add string resource for TopBar title

Abdulrahman-Ragab-01 reviewed (dismissed):
Great Job

AsmaaKhaledGomaa commented on designSystem/src/main/java/com/london/designsystem/component/TopBar.kt:
```diff
@@ -0,0 +1,149 @@
+fun TopBar(
+    type: TopBarType = TopBarType.Default,
+    title: String = stringResource(R.string.my_account),
+    onBackClick: (() -> Unit),
```
Please make `onBackClick` optional since some TopBar types don't have a back button:
`onBackClick: (() -> Unit)? = null`

AsmaaKhaledGomaa replied:
![Screenshot 2025-07-03 110423](https://github.com/user-attachments/assets/c944bf65-a51c-4f70-b5e6-8f744b732393)
![Screenshot 2025-07-03 110515](https://github.com/user-attachments/assets/e4e4ac4d-dbe1-4ddb-91f6-1e53a7d816bc)

Mostafa-alsaygh replied:
ok i will

AsmaaKhaledGomaa commented on designSystem/src/main/java/com/london/designsystem/component/TopBar.kt:
```diff
@@ -0,0 +1,149 @@
+                }
+            }
+
+            else -> {
```
Please after add null to onBackClick.
Add null safety check before calling onBackClick

AsmaaKhaledGomaa commented on designSystem/src/main/java/com/london/designsystem/component/TopBar.kt:
```diff
@@ -0,0 +1,149 @@
+import com.london.designsystem.theme.ThemePreviews
+
+@Composable
+fun TopBar(
```
Please add modifier: Modifier = Modifier

AsmaaKhaledGomaa commented on designSystem/src/main/java/com/london/designsystem/component/TopBar.kt:
```diff
@@ -0,0 +1,149 @@
+    option2: (() -> Unit)? = null
+) {
+    Row(
+        modifier = Modifier
```
Please after adding modifier you can use it here.

AsmaaKhaledGomaa commented on designSystem/src/main/java/com/london/designsystem/component/TopBar.kt:
```diff
@@ -0,0 +1,149 @@
+    }
+}
+
+enum class TopBarType {
```
I suggest removing the TopBarType enum and using individual boolean or optional parameters instead. This will make the API more flexible and easier to understand:
showBackButton: Boolean = true
showLogo: Boolean = false
actions: List<TopBarAction> = emptyList()

Mostafa-alsaygh committed:
refactor: Simplify TopBar component and introduce DefaultTopBar

This commit refactors the `TopBar` component by:
- Removing the `TopBarType` enum and its associated logic. The `TopBar` now handles variations based on the presence of `title`, `onBackClick`, `option1`, and `option2` parameters.
- Extracting the default top bar behavior into a new `DefaultTopBar` composable. This component displays the app logo, name, and a short description.

The `TopBar` component now provides a more flexible way to configure top bars by directly passing the necessary parameters. The `DefaultTopBar` handles the specific case of the application's default top bar.

Mostafa-alsaygh committed:
feat: Add theme previews annotation to top bar components

Mostafa-alsaygh commented:
> Move these files to the system design module.

Mostafa-alsaygh committed:
Merge branch 'develop' into component/top-bar

# Conflicts:
#	designSystem/src/main/res/values-ar/strings.xml
#	designSystem/src/main/res/values/strings.xml

Mostafa-alsaygh committed:
Merge branch 'develop' into component/top-bar

yusufnasserdev commented on designSystem/src/main/java/com/london/designsystem/component/DefaultTopBar.kt:
```diff
@@ -0,0 +1,57 @@
+import com.london.designsystem.theme.ThemePreviews
+
+@Composable
+fun DefaultTopBar(
+) {
```
Why not allow for an external modifier for this component? also this file can use a little formatting.

Mostafa-alsaygh replied:
Modifier for what it is not a normal component to change on it is appears

yusufnasserdev replied:
This is a component in a design system, it should be modifiable and reusable.

marwanqashwa reviewed (dismissed):
good job

Mostafa-alsaygh committed:
add modifier parameter to top bar components

Mostafa-alsaygh committed:
Merge remote-tracking branch 'origin/component/top-bar' into component/top-bar

Mostafa-alsaygh committed:
Merge branch 'develop' into component/top-bar

marwanqashwa committed:
Merge branch 'develop' into component/top-bar
